### DISCUSSION TITLE
What happens with auto updating casks

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#3359] [Casks] [answered]
Say I install a cask with homebrew, `alt-tab` for example. It has an option to `Auto-install updates periodically`. What happens when `alt-tab` updates itself ? Will homebrew recognise the version change ? Will the auto-update even work ? Will homebrew overwrite the cask once the version changes inside the tap without checking the local version ?

**Comment by SMillerDev** [comment]
If the `auto_updates` information inside the cask is correct `brew upgrade` will not attempt to update that cask unless you pass `--greedy`, in which case it will override the version.

**Reply by ghost** [reply]
Maybe my problem is that I'm new to mac os in general and I don't understand the installation process; Will the application auto-updating itself interfere in anyway shape or form with homebrew because it was installed with homebrew ? I'm familiar with Linux package managers + scoop / chocolatey on Windows and in these cases most of the time auto-updating would not work / break something.

**Reply by SMillerDev** [reply, marked as answer]
It can interfere if Homebrew doesn't know this software auto updates. Generally, it does not though.

**Reply by GwynethLlewelyn** [reply]
As far as I can tell, Homebrew always correctly figures things out on its own regarding auto-updating casks. Sometimes, that's not necessarily a "good thing" — I _like_ all my software to be regularly updated, not only when I _use_ it :) I suppose, to avoid using `--greedy` all the time, the alternative is _not_ to turn auto-updates on in the application itself, and let Homebrew figure out if it's auto-updating or not? To be very, very honest, I never tried this myself, but I found it always mysterious... without, however, ever having Homebrew break anything.

**Reply by p-linnane** [reply]
@GwynethLlewelyn Please don't necrobump 2 and a half year old threads. This discussion has run it's course.

**Reply by TehBrian** [reply]
If the discussion has run its course, why is it pinned and still marked unresolved?

**Reply by ghost** [reply]
> It can interfere if Homebrew doesn't know this software auto updates. Generally, it does not though.

I've marked this as the answer, because this has generally been my experience on Mac OS these last two years. I still try to disable auto updates for casks that I handle with hombrew.
